packages/mco/components/modals/app-manage-policies/volume-consistency-group-view.tsx

```diff
@@ -0,0 +1,294 @@
+  pvcs: string[];
+};
+
+export const VolumeConsistencyGroupView: React.FC<
+  VolumeConsistencyGroupViewProps
+> = ({ setModalContext, drInfo }) => {
+  const { t } = useCustomTranslation();
+  const [searchValue, setSearchValue] = React.useState('');
+  const [selectedFilter, setSelectedFilter] = React.useState('all');
+
+  const [mcv, mcvsLoaded, mcvsLoadError] =
+    useK8sWatchResource<ACMManagedClusterViewKind>({
+      kind: referenceForModel(ACMManagedClusterViewModel),
+      namespace: getAnnotations(drInfo.placementControlInfo[0])?.[
+        LAST_APP_DEPLOYMENT_CLUSTER_ANNOTATION
+      ],
+      name:
+        getName(drInfo.placementControlInfo[0]) +
+        '-' +
+        getNamespace(drInfo.placementControlInfo[0]) +
+        '-vrg-mcv',
+    });
+
+  const consistencyGroups: ConsistencyGroupInfo[] = React.useMemo(() => {
+    if (!mcvsLoaded || mcvsLoadError || !mcv) {
+      return [];
+    }
+
+    const groups: ConsistencyGroupInfo[] = [];
+    const vrg = mcv?.status?.result as DRVolumeReplicationGroupKind;
+
+    if (!vrg) {
+      return [];
+    }
+    const cgMap = new Map<
+      string,
+      {
+        type: string;
+        pvcs: string[];
+        namespace: string;
+      }
+    >();
+    const protectedPVCs = vrg?.status?.protectedPVCs || [];
+    protectedPVCs.forEach((pvc) => {
+      const cgLabel = Object.entries(pvc.labels || {}).find(
+        ([key]) => key === 'ramendr.openshift.io/consistency-group'
+      );
+
+      if (!cgLabel) return;
+
+      const cgName = cgLabel[1] as string;
+      const cgType = cgName.startsWith('cephfs-') ? 'Filesystem' : 'Block';
+
+      if (!cgMap.has(cgName)) {
+        cgMap.set(cgName, {
+          type: cgType,
+          pvcs: [],
+          namespace: vrg?.metadata?.namespace || '-',
+        });
+      }
+
+      const group = cgMap.get(cgName);
+      group.pvcs.push(pvc.name);
+    });
+
+    cgMap.forEach((value, cgName) => {
+      groups.push({
+        name: cgName,
+        type: value.type as 'Block' | 'Filesystem',
+        namespace: value.namespace,
+        state: vrg?.status?.state || 'Unknown',
+        lastSyncTime: vrg?.status?.lastGroupSyncTime || '-',
+        pvcs: value.pvcs,
+      });
+    });
+
+    return groups;
+  }, [mcv, mcvsLoaded, mcvsLoadError]);
+
+  const filteredGroups = React.useMemo(() => {
+    let filtered = consistencyGroups;
+
+    if (searchValue) {
+      const lowerSearch = searchValue.toLowerCase();
+      filtered = filtered.filter(
+        (group) =>
+          group.name.toLowerCase().includes(lowerSearch) ||
+          group.namespace.toLowerCase().includes(lowerSearch) ||
+          group.pvcs.some((pvc) => pvc.toLowerCase().includes(lowerSearch))
+      );
+    }
+
+    if (selectedFilter !== 'all') {
+      filtered = filtered.filter(
+        (group) => group.type.toLowerCase() === selectedFilter.toLowerCase()
+      );
+    }
+
+    return filtered;
+  }, [consistencyGroups, searchValue, selectedFilter]);
```

Comment:
```suggestion
  const filteredGroups = React.useMemo(() => 
    filterGroups(consistencyGroups, searchValue, selectedFilter
  ), [consistencyGroups, searchValue, selectedFilter]);

```

```
const filterGroups = (groups, searchValue, selectedFilter) => {
  const lowerSearch = searchValue.toLowerCase();
  return groups.filter(
    (group) =>
      (selectedFilter === 'all' || group.type.toLowerCase() === selectedFilter.toLowerCase()) &&
      (group.name.toLowerCase().includes(lowerSearch) ||
        group.namespace.toLowerCase().includes(lowerSearch) ||
        group.pvcs.some((pvc) => pvc.toLowerCase().includes(lowerSearch)))
  );
};
```
this useMemo dont have any state update, so lets keep this logic out of component